wgpu/src/lib.rs

```diff
@@ -1771,8 +1772,10 @@ impl Instance {
         }
 
         #[cfg(webgpu)]
-        if _instance_desc.backends.contains(Backends::BROWSER_WEBGPU)
-            && crate::backend::get_browser_gpu_property().map_or(false, |gpu| !gpu.is_undefined())
+        if !cfg!(wgpu_core) // If wgpu-core isn't enabled we can't use anything else.
+            || (_instance_desc.backends.contains(Backends::BROWSER_WEBGPU)
+                && crate::backend::get_browser_gpu_property()
+                    .map_or(false, |gpu| !gpu.is_undefined()))
```

Comment:
Could we split these up into 3 named variables (within the webgpu block) to make this a bit less logic soup?